pkg/resourcecollector/resourcecollector.go

```diff
@@ -42,6 +42,8 @@ const (
 	StorkResourceHash = "stork.libopenstorage.org/resource-hash"
 	// SkipModifyResources is the annotation used to skip update of resources
 	SkipModifyResources = "stork.libopenstorage.org/skip-modify-resource"
+	// ProjectMappingsOption is a resource collector option to provide rancher project mappings
+	ProjectMappingsOption = "ProjectMappings"
```

Comment:
I don't see this options being passed anywhere in migration controller when clusterpair has Rancher project mapping set, could you check if this is missed

Reply:
Good point! However that made me realize is there is another bug in the way this options are handled

- The resource collector instance is a common instance used in backup and migration controllers. A common instance cannot dictate the options used for different migration or backup objects. 
- Instead of a global map on the resource collector options we should use an actual Options object and pass it as an argument to every API.

I have made that change as a subsequent commit on this PR.